pkg/ottl/contexts/internal/ctxutil/map.go

```diff
@@ -14,21 +14,10 @@ import (
 )
 
 func GetMapValue[K any](ctx context.Context, tCtx K, m pcommon.Map, keys []ottl.Key[K]) (any, error) {
-	if len(keys) == 0 {
```

Comment:
We should probably keep this validation as it provides a more accurate error message.